python/src/cloudstorage/cloudstorage_api.py

```diff
@@ -273,6 +277,184 @@ def listbucket(path_prefix, marker=None, prefix=None, max_keys=None,
   return _Bucket(api, bucket, options)
 
 
+# pylint: disable=too-many-locals, too-many-branches, too-many-statements
+def compose(list_of_files, destination_file, files_metadata=None,
+            content_type=None, retry_params=None, _account_id=None):
+  """
+    Runs the GCS Compose on the given files.
+    Merges between 2 and 32 files into one file. Composite files may even
+      be built from other existing composites, provided that the total
+      component count does not exceed 1024. See here for details:
+      https://cloud.google.com/storage/docs/composite-objects
+  Args:
+    list_of_files: List of file name strings with no leading slashes or bucket.
+    destination_file: Path to the desired output file. Must have the bucket in the path.
+    files_metadata: Optional file metadata, see link for available options:
+                    https://cloud.google.com/storage/docs/composite-objects#_Xml
+    content_type: Used to specify the content-header of the output file.
+    retry_params: An api_utils.RetryParams for this call to GCS. If None,
+    the default one is used.
+    _account_id: Internal-use only.
+
+  Raises:
+    ValueError: If the number of files is outside the range of 2-32.
+  """
+  api = storage_api._get_storage_api(retry_params=retry_params,
+                                   account_id=_account_id)
+  """
+  Needed until cloudstorage_stub.py is updated to accept compose requests
+  TODO: When patched remove the True flow from this if.
+  """
+  if 'development' in os.environ.get('SERVER_SOFTWARE', '').lower():
+    def _temp_func(bucket, file_list, destination_file, content_type):
+      """
+        Dev server stub, remove when the dev server accepts compose requests.
+      """
+      bucket = '/' + destination_file.split('/')[1] + '/'
+      with open(destination_file, 'w', content_type=content_type) as gcs_merge:
+        for source_file in file_list:
+          try:
+            with open(bucket + source_file['Name'], 'r') as gcs_source:
+              gcs_merge.write(gcs_source.read())
+          except errors.NotFoundError as error:
+            logging.warn('File not found %s, skipping', source_file['file_name'])
+            raise error
+    compose_object = _temp_func
+  else:
+    compose_object = api.compose_object
+  file_list, _ = _validate_compose_list(
+                                      destination_file,
+                                      list_of_files,
+                                      files_metadata, 32)
+  compose_object(file_list, destination_file, content_type)
+
+def _compose_batch(list_of_files, destination_file, files_metadata=None,
```

Comment:
Here and in other places. We use 2 empty line to separate module-level methods/classes (and 1 to separate between class methods).